src/main/java/org/pac4j/dropwizard/Pac4jFactory.java

```diff
@@ -329,44 +363,84 @@ public void setSessionEnabled(boolean sessionEnabled) {
     }
 
     public Config build() {
-        Clients client = new Clients();
-        JaxRsConfig config = new JaxRsConfig();
+        // either the whole Config is built from a ConfigFactory class, or we initialize a default JaxRsConfig
+        final JaxRsConfig config;
+        if (configClass != null && configProperties != null) {
+            config = (JaxRsConfig) ConfigBuilder.build(configClass, configProperties);
+        } else {
+            config = new JaxRsConfig();
+        }
+        Clients newClients = config.getClients();
+        if (newClients == null) {
+            newClients = new Clients();
+            config.setClients(newClients);
+        }
+        if (newClients.getClients() == null) {
+            newClients.setClients(new ArrayList<>());
+        }
+        config.setClients(newClients);
```

Comment:
I think this line is redundant, if it was `null`, it is set at line 376 and if it wasn't, then it's already set